src/qos_core/src/protocol/services/boot.rs

```diff
@@ -210,6 +210,27 @@ pub struct ShareSet {
 	pub members: Vec<QuorumMember>,
 }
 
+impl FromStr for ShareSet {
+	type Err = serde_json::Error;
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		// Parse JSON -> ShareSet
+		let ss: ShareSet = serde_json::from_str(s)?;
+
+		// Sanity check: 1 <= K <= N
+		if ss.threshold == 0 || (ss.threshold as usize) > ss.members.len() {
```

Comment:
Where are we using the from str impl?
I think its confusing to have the threshold validity check here when deserializing. I would make it a separate method defined on ShareSet to validate